File: backend/cs_api/Dtos/PaginationDto.cs
namespace cs_api.Dtos
{
  public class PaginationDto<T>
  {
    public int Page { get; set; }
    public int PerPage { get; set; }
    public int PageCount { get; set; }
    public int RecordCount { get; set; }
    public bool HasData { get { return Data != null && Data.Any(); } }
    public IEnumerable<T>? Data { get; set; } = null;
        
  }
}

File: frontend/src/app/dtos/pagination.dto.ts
export class Pagination {
  constructor(response: any) {
    this.hasData = response.hasData;
    this.page = response.page;
    this.pageCount = response.pageCount;
    this.perPage = response.perPage;
    this.recordCount = response.recordCount;
  }

  public hasData: boolean = false;
  public page: number = 0;
  public pageCount: number = 0;
  public perPage: number = 0;
  public recordCount: number = 0;
}
/*

export interface Pagination {
  hasData: boolean;
  page: number;
  pageCount: number;
  perPage: number;
  recordCount: number;
}
*/